components/automate-cli/cmd/chef-automate/certRotate.go

```diff
@@ -0,0 +1,160 @@
+	certRotateCmd.PersistentFlags().BoolVar(&sshFlag.automate, "a2", false, "Automate Certificate Rotation")
+	certRotateCmd.PersistentFlags().BoolVar(&sshFlag.chefserver, "chefserver", false, "Chef Infra Server Certificate Rotation")
+	certRotateCmd.PersistentFlags().BoolVar(&sshFlag.chefserver, "cs", false, "Chef Infra Server Certificate Rotation")
+
+	certRotateCmd.PersistentFlags().StringVar(&certFlags.privateCert, "private-cert", "", "Private certificate")
+	certRotateCmd.PersistentFlags().StringVar(&certFlags.publicCert, "public-cert", "", "Public certificate")
+}
+
+func certRotate(cmd *cobra.Command, args []string) error {
+	privateCertPath := certFlags.privateCert
+	publicCertPath := certFlags.publicCert
+	fileName := "cert-rotate.toml"
+	timestamp := time.Now().Format("20060102150405")
+
+	if privateCertPath == "" || publicCertPath == "" {
+		return errors.New("Please provide public and private cert paths")
+	}
+	privateCert, err := ioutil.ReadFile(privateCertPath) // nosemgrep
+	if err != nil {
+		return status.Wrap(
+			err,
+			status.FileAccessError,
+			fmt.Sprintf("failed reading data from file: %s", err.Error()),
+		)
+	}
+
+	publicCert, err := ioutil.ReadFile(publicCertPath) // nosemgrep
+	if err != nil {
+		return status.Wrap(
+			err,
+			status.FileAccessError,
+			fmt.Sprintf("failed reading data from file: %s", err.Error()),
+		)
+	}
+
+	f, err := os.Create(fileName)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	if isA2HARBFileExist() {
+
+		infra, err := getAutomateHAInfraDetails()
+		if err != nil {
+			return err
+		}
+		sshUser := infra.Outputs.SSHUser.Value
+		sskKeyFile := infra.Outputs.SSHKeyFile.Value
+		sshPort := infra.Outputs.SSHPort.Value
+
+		if sshFlag.automate {
+			config := fmt.Sprintf(`
+			[[load_balancer.v1.sys.frontend_tls]]
+				cert = """%v"""
+				key = """%v"""
+			[[global.v1.frontend_tls]]
+				cert = """%v"""
+				key = """%v"""
+		`, string(publicCert), string(privateCert), string(publicCert), string(privateCert))
+
+			_, err = f.Write([]byte(config))
+			if err != nil {
+				log.Fatal(err)
+			}
+			f.Close()
+			automateIps := infra.Outputs.AutomatePrivateIps.Value
+			if len(automateIps) == 0 {
+				return errors.New("No Automate Ips found")
+			}
+
+			const remoteService string = "automate"
+			scriptCommands := fmt.Sprintf(FRONTEND_COMMANDS, remoteService+timestamp, dateFormat)
+			for i := 0; i < len(automateIps); i++ {
+				err := copyFileToRemote(sskKeyFile, fileName, sshUser, automateIps[i], remoteService+timestamp)
+				if err != nil {
+					writer.Errorf("%v", err)
+					return err
+				}
+				output, err := ConnectAndExecuteCommandOnRemote(sshUser, sshPort, sskKeyFile, automateIps[i], scriptCommands)
+				if err != nil {
+					writer.Errorf("%v", err)
+					return err
+				}
+				writer.Printf(output)
+			}
+
+		} else if sshFlag.chefserver {
+			config := fmt.Sprintf(`
+			[[load_balancer.v1.sys.frontend_tls]]
```

Comment:
Repetitive strings> Assign this string to a variable and use that variable in both places (automate & chef-server)